DesignSafe-CI/portal
# task/DES-2702: Tapis v3 Auth

## Overview: ##

Adds `tapipy` package, Tapis v3 OAuth Token model, and v3 authorization flow

## PR Status: ##


## Related Jira tickets: ##

* [DES-2702](https://tacc-main.atlassian.net/browse/DES-2702)

## Testing Steps: ##
1. Update `designsafe.env` with https://stache.utexas.edu/entry/ee36851448345135988a15c0684464ce to get new `TAPIS_` settings
2. `make build && make start`
3. `./manage.py migrate` in `des_django`
4. Navigate to https://designsafe.dev/login/ and confirm login works

## UI Images:
You should be redirected to the Tapis OAuth login screen:
![Screenshot 2024-02-28 at 3 40 44 PM](https://github.com/DesignSafe-CI/portal/assets/20326896/e94c7abc-6242-4846-bb59-d6aa418fcc2c)

Then after successful login, redirected back to the profile:
![Screenshot 2024-02-28 at 3 42 19 PM](https://github.com/DesignSafe-CI/portal/assets/20326896/0a37c49b-1166-4f9f-998e-36c8f099273c)


## Notes: ##
This includes breaking changes to the `main` branch. I propose we point this to a feature branch for v3 development.
